plugins/LayerView/LayerView.qml

```diff
@@ -114,4 +145,129 @@ Item
             }
         }
     }
+
+    Rectangle {
+        anchors.left: parent.left
+        anchors.verticalCenter: parent.verticalCenter
+        anchors.top: slider_background.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+        //anchors.leftMargin: UM.Theme.getSize("default_margin").width
+        width: UM.Theme.getSize("slider_layerview_background").width * 3
+        height: slider.height + UM.Theme.getSize("default_margin").height * 2
+        color: UM.Theme.getColor("tool_panel_background");
+        border.width: UM.Theme.getSize("default_lining").width
+        border.color: UM.Theme.getColor("lining")
+
+        ListModel
+        {
+            id: layerViewTypes
+            ListElement {
+                text: "Material color"
+                type_id: 0
+            }
+            ListElement {
+                text: "Line type"
+                type_id: 1  // these ids match the switching in the shader
+            }
+        }
+
+        ComboBox
+        {
+            id: layerTypeCombobox
+            anchors.top: parent.top
+            anchors.left: parent.left
+            model: layerViewTypes
+            visible: !UM.LayerView.compatibilityMode
+            onActivated: {
+                UM.LayerView.setLayerViewType(layerViewTypes.get(index).type_id);
+            }
+        }
+
+        Label
+        {
+            id: compatibilityModeLabel
+            anchors.top: parent.top
+            anchors.left: parent.left
+            text: catalog.i18nc("@label","Compatibility mode")
+            visible: UM.LayerView.compatibilityMode
+        }
+
+        ColumnLayout {
+            id: view_settings
+            anchors.top: UM.LayerView.compatibilityMode ? compatibilityModeLabel.bottom : layerTypeCombobox.bottom
+            anchors.topMargin: UM.Theme.getSize("default_margin").height
+            x: UM.Theme.getSize("default_margin").width
+
+            CheckBox {
```

Comment:
This would be better with a repeater & a listModel. This is a whole lot of code duplication.